### PR TITLE
keyformat, internal: add fuzzers

### PR DESCRIPTION
**Pull request by odeke-em**
Adds fuzzers that have discovered some bugs, like:
* https://github.com/cosmos/iavl/issues/760
* https://github.com/cosmos/iavl/issues/761

/cc @elias-orijtech @tac0turtle

### CODE REVIEW

File: internal/rand/fuzz_test.go [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+	// 2. Now run the fuzzer.
+	rr := NewRand()
+	f.Fuzz(func(t *testing.T, n int) {
+		s := rr.Str(n)
+		// Ensure that we've got alphanumeric sequences entirely.
+		for _, c := range s {
+			switch {
+			case c >= '0' && c <= '9':
+			case c >= 'a' && c <= 'z':
+			case c >= 'A' && c <= 'Z':
```
Comment (tac0turtle) [inline internal/rand/fuzz_test.go:28]: please delete, we dont need to fuzz test helpers

Comment (tac0turtle) [reply internal/rand/fuzz_test.go:28]: ping

Comment (odeke-em) [reply internal/rand/fuzz_test.go:28]: Sure, I'll remove it.

**Comment by odeke-em** [comment]
Closing this PR since RandStr is deemed to be an internal function only used in tests.

**Pull request closed without merging**